community/modules/compute/schedmd-slurm-gcp-v5-partition/main.tf

```diff
@@ -69,6 +69,16 @@ locals {
   # Also, slurm imposed a lot of restrictions to this name, so we format it to an acceptable string
   tmp_cluster_name   = substr(replace(lower(var.deployment_name), "/^[^a-z]*|[^a-z0-9]/", ""), 0, 8)
   slurm_cluster_name = var.slurm_cluster_name != null ? var.slurm_cluster_name : local.tmp_cluster_name
+
+  uses_zone_policies = length(var.zone_policy_allow) + length(var.zone_policy_deny) > 0
+  excluded_zones     = var.zone == null ? [] : [for z in data.google_compute_zones.available.names : z if z != var.zone]
+  zone_policy_deny   = local.uses_zone_policies ? var.zone_policy_deny : local.excluded_zones
+  zone_policy_allow  = local.uses_zone_policies || var.zone == null ? var.zone_policy_allow : [var.zone]
```

Comment:
I'll admit to not loving that the pattern for these 2 lines differs since you're really sneaking in the default value for var.zone_policy_allow when var.zone is null. Can you just add a comment at line 72 which explains the logic? e.g.

```
# use var.zone_policy_allow and var.zone_policy_deny if either or both are set
# if neither are set and var.zone is unset: choose empty list for deny and allow
# if neither are set and var.zone is unset: choose empty list for deny and [var.zone] for allow
# use var.zone_policy_allow in all cases except when var.zone is set
```


FYI: this is a good case where you would want `nullable = false` for these 2 variables because then you don't have to worry about them being `null`. Right now, you could pass an explicit null and it would break. I do not think you should develop against that scenario, though.

Reply:
These 3 comments are all right around nit level. I am going to collect them in a task and approve PR.